sdk/client.py

```diff
@@ -147,18 +43,19 @@ def __init__(
         token_provider: Optional[TokenProvider] = None,
         endpoints: Optional[Endpoints] = None,
     ):
-        if not (username and password) and not token_provider:
-            raise Exception(
-                "At least a username/password combination or"
-                "TokenProvider object should be provided."
+        if not username and not token_provider:
+            raise ValueError(
+                "At least a username or TokenProvider object should be provided."
             )
+        
+        if password is None and username is not None:
```

Comment:
could you make this
```
if username and not password:
```
?
would catch an empty string too, which is more convenient for env var setups

Reply:
Line 46-49, just above, the code checks it already:

```python
        if not username and not token_provider:
            raise ValueError(
                "At least a username or TokenProvider object should be provided."
            )
```

Reply:
what if I input

```
password = ""
username = "my_username"
```

This would not be caught.

Reply:
I added a test to prove it would be catch, see `test_module_no_user_with_password`

Reply:
Thanks.  I agree the current implementation will not yield an invalid instance of the SDK.

What I was actually concerned about was this:
```
$ export MY_ENV_VAR=
$ python3
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> a = os.environ["MY_ENV_VAR"]
>>> a is None
False
>>> type(a)
<class 'str'>
>>> 
```

It might seem artificial to do `export MY_ENV_VAR=`, however, this can easily happen in practice. Especially when the sdk is used as part of an application or a different library.

Which is why, when we're first adding the `getpass` functionality, it would be great for empty password to also be caught in that functionality. If, instead of testing `if password is None` we just test `if password`, this is caught.

Sorry for being vague before, and for laboring on this point, but I think it will save us some time in the future.

Reply:
Thanks! Your latest changes look like a great improvement!
I still had one minor nit (arguably also personal preference).